File: src/main/java/com/core/base/corebase/common/exception/BaseException.java
package com.core.base.corebase.common.exception;

import com.core.base.corebase.common.code.ErrorCode;
import lombok.Getter;

@Getter
public class BaseException extends RuntimeException {
    private final ErrorCode errorCode;
    private final String reason;

    public BaseException(ErrorCode errorCode) {
        this(errorCode, null);
    }

    public BaseException(ErrorCode errorCode, Object value) {
        this.errorCode = errorCode;
        if (value != null) {
            this.reason = errorCode.getReason() + " " +
                         errorCode.getDetail().replace("{VALUE}", value.toString());
        } else {
            this.reason = errorCode.getReason();
        }
    }
}

File: src/main/kotlin/com/core/base/corebase/common/exception/BaseException.kt
package com.core.base.corebase.common.exception

import com.core.base.corebase.common.code.ErrorCode

class BaseException (
    val errorCode: ErrorCode,
    value: Any? = null
) : RuntimeException() {
    val reason: String =
        if (value != null) ("${errorCode.reason} ${errorCode.detail.replace("{VALUE}", value.toString())}")
        else errorCode.reason
}